src/dbally/gradio/gradio_interface.py

```diff
@@ -0,0 +1,181 @@
+
+        Returns:
+            A tuple containing the preview dataframe and load status text.
+        """
+        selected_view = self.collection.get(selected_view_name)
+        text_to_display = "No data preview available"
+        if issubclass(type(selected_view), BaseStructuredView):
+            selected_view_results = selected_view.execute()
+            preview_dataframe = pd.DataFrame.from_records(selected_view_results.results).head(self.preview_limit)
+            text_to_display = "Data preview loaded"
+        else:
+            preview_dataframe = pd.DataFrame()
+
+        return preview_dataframe, text_to_display
+
+    async def _ui_ask_query(self, question_query: str) -> Tuple[Dict, Optional[pd.DataFrame], str]:
+        """
+        Asynchronously processes a query and returns the results.
+
+        Args:
+            question_query (str): The query to process.
+
+        Returns:
+            A tuple containing the generated query context, the query results as a dataframe, and the log output.
+        """
+        self.log.seek(0)
+        self.log.truncate(0)
+        try:
+            execution_result = await self.collection.ask(question=question_query)
+            generated_query = str(execution_result.context)
+            data = pd.DataFrame.from_records(execution_result.results)
+        except UnsupportedQueryError:
+            generated_query = {"Query": "unsupported"}
+            data = pd.DataFrame()
+        except NoViewFoundError:
+            generated_query = {"Query": "No view matched to query"}
+            data = pd.DataFrame()
+        except PromptTemplateError:
+            generated_query = {"Query": "No view matched to query"}
+            data = pd.DataFrame()
+        finally:
+            self.log.seek(0)
+            log_content = self.log.read()
+        return generated_query, data, log_content
+
+    async def create_interface(
+        self, user_collection: Collection, preview_limit: int = 20
+    ) -> Optional[gradio.Interface]:
+        """
+        Creates a Gradio interface for interacting with the user collection and similarity stores.
+
+        Args:
+            user_collection: The user's collection to interact with.
+            preview_limit: The maximum number of preview data records to display. Default is 20.
+
+        Returns:
+            The created Gradio interface, or None if no data is available to load.
+
+        Raises:
+             ValueError: occurs when there is no view define in collection.
+        """
+
+        self.preview_limit = preview_limit
+        self.collection = user_collection
+        self.collection.add_event_handler(CLIEventHandler(self.log))
+
+        view_list = [*user_collection.list()]
+        if view_list:
+            default_selected_view_name = view_list[0]
+        else:
+            raise ValueError("No view to display")
+
+        if not view_list:
```

Comment:
Above you raise an exception if `view_list` is empty, so it's not needed here. Maybe we shouldn't really do anything if there is no view to display, just render an empty boilerplate.